html5ever/src/tree_builder/mod.rs

```diff
@@ -1392,7 +1414,7 @@ where
         // template start tag's shadowrootmode is not in the none state
         let is_shadow_root_mode = tag.attrs.iter().any(|attr| {
             attr.name.local == local_name!("shadowrootmode")
-                && (attr.value.to_string() == *"open" || attr.value.to_string() == *"close")
+                && (attr.value.to_string() == *"open" || attr.value.to_string() == *"closed")
```

Comment:
As noted in https://github.com/servo/html5ever/pull/568#discussion_r1918059588 you can do these comparisons without allocating a new string